packages/editor/src/store/selectors.js

```diff
@@ -659,7 +682,42 @@ export const getBlock = createSelector(
 	},
 	( state, clientId ) => [
 		state.editor.present.blocks.byClientId[ clientId ],
+		state.editor.present.blocks.attributesByClientId[ clientId ],
 		getBlockDependantsCacheBust( state, clientId ),
+		state.editor.present.blocks.order,
+		state.editor.present.edits.meta,
+		state.initialEdits.meta,
+		state.currentPost.meta,
+	]
+);
+
+/**
+ * Returns a block given its client ID. This is a parsed copy of the block,
+ * containing its `blockName` and `clientId`. This
+ * is not the block's registration settings, which must be retrieved from the
+ * blocks module registration store.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block client ID.
+ *
+ * @return {Object} Parsed block object.
+ */
+export const getBlockWithoutAttributes = createSelector(
```

Comment:
I'm not certain why we need all these new selectors `getBlockWithoutAttributes` `getBlocksWithoutAttributes`, `getBlocksByClientIdWithoutAttributes`. I'd like to avoid exposing new selectors in this PR. Is this only for `getBlocksForSerializationWithoutAttributes`?

Reply:
It gets used directly or indirectly in a bunch of other selectors (such as `getBlockNamesForSerialization`, `getInserterItems`, etc.) , but you're right that we don't need to export these new selectors.

Reply:
The latest commit removes the new selectors, in exchange for the approach of using the same full selectors everywhere, but changing the dependencies where possible. This way, they won't invalidate on attribute changes, but that's ok if the attributes are not being used.